specs/Standards/MultiToken/Core.md

```diff
@@ -0,0 +1,355 @@
+// The balance though wrapped in quotes and treated like a string, 
+// the number will be stored as an unsigned integer with 128 bits.
+// Arguments:
+// * `account_id`: the NEAR account that owns the token.
+// * `token_id`: the token to retrieve the balance from
+function mt_balance_of(account_id: string, token_id: string): string
+
+// Returns the balances of an account for the given `token_ids`.   
+// The balances though wrapped in quotes and treated like strings, 
+// the numbers will be stored as an unsigned integer with 128 bits.
+// Arguments:
+// * `account_id`: the NEAR account that owns the tokens.
+// * `token_ids`: the tokens to retrieve the balance from
+function mt_batch_balance_of(account_id: string, token_ids: string): string[]
+
+// Returns the token supply with the given `token_id` or `null` if no such token exists.
+// The supply though wrapped in quotes and treated like a string, the number will be stored 
+// as an unsigned integer with 128 bits.
+function mt_supply(token_id: string): string | null 
+
+// Returns the token supplies with the given `token_ids`, a string value is returned or `null` 
+// if no such token exists. The supplies though wrapped in quotes and treated like strings, 
+// the numbers will be stored as an unsigned integer with 128 bits.
+function mt_batch_supply(token_ids: string[]): (string | null)[]
+```
+
+The following behavior is required, but contract authors may name this function something other than the conventional `mt_resolve_transfer` used here.
+
+```ts
+// Finalize an `mt_transfer_call` or `mt_batch_transfer_call` chain of cross-contract calls. Generically 
+// referred to as `mt_transfer_call` as it applies to `mt_batch_transfer_call` as well.
+//
+// The `mt_transfer_call` process:
+//
+// 1. Sender calls `mt_transfer_call` on MT contract
+// 2. MT contract transfers token from sender to receiver
+// 3. MT contract calls `mt_on_transfer` on receiver contract
+// 4+. [receiver contract may make other cross-contract calls]
+// N. MT contract resolves promise chain with `mt_resolve_transfer`, and may
+//    transfer token back to sender
+//
+// Requirements:
+// * Contract MUST forbid calls to this function by any account except self
+// * If promise chain failed, contract MUST revert token transfer
+// * If promise chain resolves with `true`, contract MUST return token to
+//   `sender_id`
+//
+// Arguments:
+// * `sender_id`: the sender of `mt_transfer_call`
+// * `receiver_id`: the `receiver_id` argument given to `mt_transfer_call`
+// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+// * `amounts`: the `token_ids` argument given to `mt_transfer_call`
+// * `approved_token_ids`: if using Approval Management, contract MUST provide
+//   set of original approved accounts in this argument, and restore these
+//   approved accounts in case of revert.
+//
+// Returns total amount spent by the `sender_id`, corresponding to the `token_id`.
```

Comment:
I'm not too sure what it means by the total amount spent by the sender_id? Isn't the transfer call just saying what amounts to transfer for the given token_id? If it's an NFT, the amounts can never be above 1 correct? similarly if you have a FT and you want to transfer 100, you would just pass in the FT token_id and then an amount of 100. I'm a bit confused as to the "spent" part

Reply:
transfer_call is just saying the amounts to tranfser for given_token_id.  Correct see secondary comment. Sender should say receiver_id